lib/semantics/check-select-stmt.cc

```diff
@@ -0,0 +1,121 @@
+//==check-select-stmt.cc - Checker for select-rank
+// TODO:select-case
+// TODO:select-type
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===--------------------------------------------------------------------------------------===
+
+#include "check-select-stmt.h"
+#include "tools.h"
+#include "../common/Fortran.h"
+#include "../common/idioms.h"
+#include "../parser/message.h"
+#include "../parser/tools.h"
+#include <list>
+#include <optional>
+#include <set>
+#include <tuple>
+#include <variant>
+
+namespace Fortran::semantics {
+
+void SelectStmtChecker::Leave(
+    const parser::SelectRankConstruct &selectRankConstruct) {
+  const auto &selectRankStmt{
+      std::get<parser::Statement<parser::SelectRankStmt>>(
+          selectRankConstruct.t)};
+  const auto &selectRankStmtSel{
+      std::get<parser::Selector>(selectRankStmt.statement.t)};
+  // R1149 select-rank-stmt checks
+  std::visit(
```

Comment:
C1150 requires the *selector* to be the name of an assumed-rank dummy argument, so this code can be simplified and completed.  Use `UnwrapWholeSymbolDataRef()` from `lib/evaluate/tools.h` to determine whether the selector is a simple name reference; if it is, verify that it's an assumed-rank dummy argument, and if it isn't, emit an error.